inference-engine/src/vpu/graph_transformer/src/stages/ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,173 @@
+                                                   int32_t blankIndex) {
+    auto stage = model->addNewStage<CTCGreedyDecoderSeqLenStage>(name,
+                                                                 StageType::CTCGreedyDecoderSeqLen,
+                                                                 layer, inputs, outputs);
+    stage->attrs().set<bool>("mergeRepeated", mergeRepeated);
+    stage->attrs().set<int32_t>("blankIndex", blankIndex);
+
+    return stage;
+}
+
+void FrontEnd::parseCTCGreedyDecoderSeqLen(const Model& model, const ie::CNNLayerPtr& layer,
+                                           const DataVector& inputs, const DataVector& outputs) const {
+    VPU_THROW_UNLESS(layer, "CNNLayer pointer is null.");
+    VPU_THROW_UNLESS(inputs.size() == 2 || inputs.size() == 3,
+                     "{} layer with name {} must have 2 or 3 inputs, actually "
+                     "provided {} inputs",
+                     layer->type, layer->name, inputs.size());
+    VPU_THROW_UNLESS(outputs.size() == 2,
+                     "{} layer with name {} must have 2 outputs, actually "
+                     "provided {} outputs",
+                     layer->type, layer->name, outputs.size());
+
+
+    const auto mergeRepeated = layer->GetParamAsBool("merge_repeated");
+    const auto blankIndex = [&] {
+        if (inputs.size() == 3) {
+            VPU_THROW_UNLESS(inputs[2]->usage() == DataUsage::Const,
+                             "Only constant axis is supported, but got {} data object",
+                             inputs[2]->usage());
+            VPU_THROW_UNLESS(inputs[2]->desc().totalDimSize() == 1,
+                             "Only single value blankIndex is supported, got {} elements",
+                             inputs[2]->desc().totalDimSize());
+
+            return *inputs[2]->content()->get<int32_t>();
+        }
+
+        const auto classes = inputs[0]->desc().dim(Dim::W);
+        return classes - 1;
+    }();
+
+    const auto toUpper = [](const std::string& str) {
+        std::string result;
+        result.reserve(str.size());
+        std::transform(begin(str), end(str), std::back_inserter(result),
+                       [](char c) {return std::toupper(c);} );
+        return result;
+    };
+
+    const auto classesIndexType = toUpper(layer->GetParamAsString("classes_index_type"));
+    const auto sequenceLengthType = toUpper(layer->GetParamAsString("sequence_length_type"));
+
+    VPU_THROW_UNLESS(classesIndexType == "I32", "classes_index_type == %s. Only I32 is supported",
+                     classesIndexType);
+
+    VPU_THROW_UNLESS(sequenceLengthType == "I32", "sequence_length_type == %s. Only I32 is supported",
+                     sequenceLengthType);
+
+    _stageBuilder->addCTCGreedyDecoderSeqLenStage(model, layer->name, layer,
+                                                  inputs, outputs, mergeRepeated, blankIndex);
```

Comment:
looks strange that lambda expression is passed as int parameter. 
Does the compiler convert it to call `blankIndex()` automatically?

Reply:
Lambda is being immediately invoked in place. `blankIndex` has type `int`.
https://github.com/openvinotoolkit/openvino/blob/14a0e654dc8cbe1909f7c10c801aba32d341c2d4/inference-engine/src/vpu/graph_transformer/src/stages/ctc_greedy_decoder_seq_len.cpp#L147-L151